### PR TITLE
Trips validation and drop downs

### PR DESCRIPTION
**Pull request by amhursh**
Added ability to select start and end station id when creating a trip and implemented drop down menus to ensure appropriate selection. Added validation accordingly and refactored tests to get things passing.

### CODE REVIEW

File: app/views/trips/new.erb [hunk 1]
```diff
@@ -20,19 +20,39 @@
...
+        <option value="" disabled selected>Name of the station trip started at</option>
+        <% @stations.each do |station| %>
+        <option name="trip[start_station_id]"><%= station.name %></option>
+        <% end %>
+      </select>
     </div>
     <div class="col-md-6 mb-1">
       <label for="validationCustom03">Start Station ID: </label>
-      <input type="text" class="form-control" name="trip[start_station_id]" placeholder="ID of the station trip started at"/>
+      <select type="text" class="form-control" id="start-id-drop-down" name="trip[start_station_id]" placeholder="ID of the station trip started at">
```
Comment (snayrouz) [inline app/views/trips/new.erb:32]: Great attention to detail

File: spec/features/trips/user_can_access_new_trip_page_spec.rb [hunk 2]
```diff
@@ -14,21 +14,34 @@
...
     fill_in("trip[bike_id]", with: 299)
     fill_in("trip[subscription_type]", with: "Subscriber")
     fill_in("trip[zip_code]", with: "90210")
     click_on("Create New Trip")
+    save_and_open_page
 
     expect(page).to have_content("173")
-    expect(page).to have_content("Yoyo")
-    expect(page).to have_content("Memememe")
+    expect(page).to have_content("Panera")
```
Comment (snayrouz) [inline spec/features/trips/user_can_access_new_trip_page_spec.rb:43]: Turing > Panera

File: spec/features/trips/user_can_access_new_trip_page_spec.rb [hunk 3]
```diff
@@ -14,21 +14,34 @@
...
+    station1 = Station.create(name: "Panera",
+                   dock_count: 15,
+                   city: "Denver",
+                   installation_date: "2014-04-09")
+    station2 = Station.create(name: "Turing",
+                   dock_count: 15,
+                   city: "Denver",
+                   installation_date: "2014-04-09")
+    # require 'pry'; binding.pry
+
```
Comment (mimizotti) [inline spec/features/trips/user_can_access_new_trip_page_spec.rb:26]: Please don't forget to delete this pry.

File: app/views/trips/new.erb (hunk 1 above)
Comment (dphilla) [reply app/views/trips/new.erb:32]: great work with the placeholders

File: app/views/trips/new.erb [hunk 4]
```diff
@@ -20,19 +20,39 @@
...
     <div class="col-md-6 mb-1">
       <label for="validationCustom05">End Station ID: </label>
-      <input type="text" class="form-control" name="trip[end_station_id]" placeholder="ID of the station trip ended at"/>
+      <select type="text" class="form-control" id="end-id-drop-down" name="trip[end_station_id]" placeholder="ID of the station trip ended at">
+        <option value="" disabled selected>ID of the station trip started at</option>
+        <% @stations.each do |station| %>
+        <option name="trip[end_station_id]"><%= station.id %></option>
+        <% end %>
+      </select>
     </div>
```
Comment (mimizotti) [inline app/views/trips/new.erb:56]: This is really clever. Good job!

**Review by mimizotti — APPROVED** [review]

**Pull request merged**